typescript/mainLogic/services/service_context.ts

```diff
@@ -0,0 +1,157 @@
+    
+    client: core_api_bindings_js_grpc.ApiContainerServiceClient;
+    serviceId: ServiceID;
+	ipAddress: string;
+	testVolumeMountpointOnTestsuiteContainer: string;
+	testVolumeMountpointOnServiceContainer: string;
+
+    constructor(
+        client: core_api_bindings_js_grpc.ApiContainerServiceClient,
+		serviceId: ServiceID,
+		ipAddress: string,
+		testVolumeMountpointOnTestsuiteContainer: string,
+		testVolumeMountpointOnServiceContainer: string) {
+            this.client = client;
+            this.serviceId = serviceId;
+            this.ipAddress = ipAddress;
+            this.testVolumeMountpointOnTestsuiteContainer = testVolumeMountpointOnTestsuiteContainer;
+            this.testVolumeMountpointOnServiceContainer = testVolumeMountpointOnServiceContainer;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public GetServiceID() {
+        return this.serviceId;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public GetIPAddress() {
+        return this.ipAddress;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public ExecCommand(command: string[]) {
+        let serviceId = this.serviceId;
+        let args = new core_api_bindings_js.ExecCommandArgs();
+        args.setServiceId(serviceId);
+        args.setCommandArgsList(command);
+
+        let resp, err = this.client.execCommand(args); //TODO - no context needed in typescript, so what should I do here?
+        if (err != null) {
+            return [0, null, console.error(
+                err,
+			    "An error occurred executing command '%v' on service '%v'",
+			    command,
+			    serviceId
+            )]; //TODO cannot return multipe values in typescript function, need to wrap in array or object
+        }
+        return resp.ExitCode, resp.LogOutput, null;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public GenerateFiles(filesToGenerateSet: Map<string, boolean>) {
+        let serviceId = this.serviceId;
+        let fileGenerationOpts = new Map(); //Key: string & Value: core_api_bindings.FileGenerationOptions{} - type not needed in typescript
+        for (let fileId in filesToGenerateSet) {
+            fileGenerationOpts[fileId] = new core_api_bindings_js.FileGenerationOptions();
+            fileGenerationOpts[fileId].setFileTypeToGenerate(core_api_bindings_js.FileGenerationOptions.FileTypeToGenerate); //TODO - Check this
+        }
+
+        let args = new core_api_bindings_js.GenerateFilesArgs();
+        args.setServiceId(String(serviceId));
+        args.getFilesToGenerateMap(fileGenerationOpts); //TODO - No option to set FilesToGenerateMap (should this be the case)??
+
+        let resp, err = this.client.generateFiles(args) //TODO TODO - don't need context, what to do?
```

Comment:
You can just omit the contexts - `contexts` are actually something fairly unique (and cool) to Go; I'd never seen them in another language